src/Core/Analyser/EventHandler/ViolationHandler.php

```diff
@@ -4,63 +4,36 @@
 
 namespace Qossmic\Deptrac\Core\Analyser\EventHandler;
 
-use Qossmic\Deptrac\Contract\Analyser\PostProcessEvent;
 use Qossmic\Deptrac\Contract\Analyser\ProcessEvent;
-use Qossmic\Deptrac\Contract\Result\Error;
+use Qossmic\Deptrac\Contract\Result\Result;
 use Qossmic\Deptrac\Contract\Result\SkippedViolation;
 use Qossmic\Deptrac\Contract\Result\Violation;
+use Qossmic\Deptrac\Core\Layer\LayerProvider;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 
-use function sprintf;
-
 /**
  * @internal
  */
-class ViolationHandler implements EventSubscriberInterface
+abstract class ViolationHandler implements EventSubscriberInterface
```

Comment:
Just a matter of personal taste, but I would prefer for this to be a dedicated service, which we inject into the event listeners instead of an abstract class.

Reply:
Makes sense, we could also expose is as part of contract for extension developers to use.